src/main/java/uk/gov/pay/products/resources/PaymentResource.java

```diff
@@ -46,6 +46,14 @@ public PaymentResource(PaymentFactory paymentFactory, PaymentRequestValidator re
         this.paymentFactory = paymentFactory;
         this.requestValidator = requestValidator;
     }
+    
+    @Path("/v1/api/payments/redact-reference/{govukPaymentId}")
+    @POST
+    public Response redactReference(@PathParam("govukPaymentId") String govukPaymentId) {
+        logger.info("Redacting reference for payment with govukPaymentId - [ {} ]", govukPaymentId);
```

Comment:
We could have some structured logging here that adds `payment_external_id` from https://github.com/alphagov/pay-java-commons/blob/f2fe233c2eb127c36f1122ae192ac0a4575d9bb8/logging/src/main/java/uk/gov/service/payments/logging/LoggingKeys.java#L165.